packages/perseus-editor/src/components/graph-points-count-selector.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from "react";
+
+const UNLIMITED = "unlimited" as const;
+type PointValue = number | typeof UNLIMITED;
+
+const GraphPointsCountSelector = ({
+    numPoints = 1,
+    onChange,
+}: {
+    numPoints?: PointValue;
+    onChange: (points: PointValue) => void;
+}) => {
+    return (
+        <select
+            key="point-select"
+            value={numPoints}
+            onChange={(e) => {
+                // Convert numbers, leave UNLIMITED intact:
+                const num = +e.target.value || UNLIMITED;
```

Comment:
This wasn't super clear to me and I'm worried it might be prone to a bug. What I interpret as happening:

1. `value` can be a number or "UNLIMITED"
2. Try to coerce value to a number
3. If it doesn't work because it's UNLIMITED, `+value` will be `NaN`
4. `NaN` is falsy, so `NaN || UNLIMITED === UNLIMITED`

Readability: I think this is pretty opaque. It's using coercion (`+`), then banks on the fact that a failed coercion is NaN and thus falsy, and then it uses the `||` operator. All pretty standard in JS, but combined makes it tough to follow.

Possible bug: below you're making a range of numbers, `[...Array(7).keys()]`. I could be wrong, but doesn't that mean it's [0, 1...5, 6]? So you select "0 points", `+"0" === 0`, `0 || UNLIMITED === UNLIMITED`. Maybe that's intentional though? If so, I think it's not particularly clear.

Reply:
This is super old code that I just brought back. But your points are valid so I'll create a small PR that extracts this into a formal function and adds a few tests.  That'll make things much clearer and less implicit.

Reply:
See #559